provider/cluster_autoscaler_resource.go

```diff
@@ -0,0 +1,703 @@
+						Type:       types.StringType,
+						Optional:   true,
+						Validators: FloatRangeValidators("utilization threshold validation", 0.0, 1.0),
+					},
+					"delay_after_add": {
+						Description: "How long after scale up that scale down evaluation resumes.",
+						Type:        types.StringType,
+						Optional:    true,
+						Validators:  DurationStringValidators("delay after add validation"),
+					},
+					"delay_after_delete": {
+						Description: "How long after node deletion that scale down evaluation resumes.",
+						Type:        types.StringType,
+						Optional:    true,
+						Validators:  DurationStringValidators("delay after delete validation"),
+					},
+					"delay_after_failure": {
+						Description: "How long after scale down failure that scale down evaluation resumes.",
+						Type:        types.StringType,
+						Optional:    true,
+						Validators:  DurationStringValidators("delay after failure validation"),
+					},
+				}),
+			},
+		},
+	}, nil
+}
+
+func RangeValidators() []tfsdk.AttributeValidator {
+	return []tfsdk.AttributeValidator{
+		&common.AttributeValidator{
+			Desc: "max must be greater or equal to min",
+			Validator: func(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+				resourceRange := &AutoscalerResourceRange{}
+				diag := req.Config.GetAttribute(ctx, req.AttributePath, resourceRange)
+				if diag.HasError() {
+					// No attribute to validate
+					return
+				}
+
+				steps := []string{}
+				for _, step := range req.AttributePath.Steps() {
+					steps = append(steps, fmt.Sprintf("%s", step))
+				}
+				if resourceRange.Min.Value > resourceRange.Max.Value {
+					resp.Diagnostics.AddAttributeError(
+						req.AttributePath,
+						"Invalid resource range",
+						fmt.Sprintf("In '%s' attribute, max value must be greater or equal to min value", strings.Join(steps, ".")),
+					)
+				}
+			},
+		},
+	}
+}
+
+func (t *ClusterAutoscalerResourceType) NewResource(ctx context.Context,
+	p tfsdk.Provider) (tfsdk.Resource, diag.Diagnostics) {
+	// Cast the provider interface to the specific implementation:
+	// use it directly when needed.
+	parent := p.(*Provider)
+
+	// Get the collection of clusters:
+	collection := parent.connection.ClustersMgmt().V1().Clusters()
+
+	// Create the resource:
+	return &ClusterAutoscalerResource{collection: collection}, nil
+}
+
+func (r *ClusterAutoscalerResource) Create(ctx context.Context,
+	request tfsdk.CreateResourceRequest, response *tfsdk.CreateResourceResponse) {
+	plan := &ClusterAutoscalerState{}
+	diags := request.Plan.Get(ctx, plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resource := r.collection.Cluster(plan.Cluster.Value)
+	// We expect the cluster to exist. Return an error if it's not the case
+	if resp, err := resource.Get().SendContext(ctx); err != nil && resp.Status() == http.StatusNotFound {
```

Comment:
Last version we changed the `WaitTillClusterReady` function to support that case